device/src/bin/frostypede.rs

```diff
@@ -445,17 +451,34 @@ where
 
         if let Workflow::UserPrompt(prompt) = &self.workflow {
             if self.select_button.is_low().unwrap() {
-                let ui_event = match prompt {
-                    Prompt::KeyGen(_) => UiEvent::KeyGenConfirm(self.user_confirm),
-                    Prompt::Signing(_) => UiEvent::SigningConfirm(self.user_confirm),
-                };
-                event = Some(ui_event);
-            } else if self.toggle_button.is_high().unwrap() {
-                self.toggled = false;
-            } else if self.toggle_button.is_low().unwrap() && !self.toggled {
-                self.user_confirm = !self.user_confirm;
-                self.toggled = true;
-                self.changes = true;
+                match self.confirm_state.poll() {
+                    AnimationProgress::Progress(progress) => {
+                        self.led
+                            .write(brightness(
+                                [RGB::new(0, (128.0 * progress) as u8, 0)].iter().cloned(),
+                                30,
+                            ))
+                            .unwrap();
+                        self.display.confirm_bar(progress).unwrap();
```

Comment:
An alternative to doing this here is to update the state of `prompt` by adding some `progress` field to it and letting render handle dong the progress bar. This would be more inline with the design but this is also more efficient so I'm not really bothered.